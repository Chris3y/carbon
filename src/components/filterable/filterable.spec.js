import React from 'react';
import { shallow } from 'enzyme';
import Filterable from './filterable.component';
import FilterableContext from './filterable.context';

describe('Filterable', () => {
  const shallowRender = props => (
    shallow(
      <Filterable { ...props }>
        test children
      </Filterable>
    )
  );

  it('defaults filterType to match', () => {
    const wrapper = shallowRender();
    expect(wrapper.props().value.filterType).toEqual('match');
  });

  it('provides a filterable context with filter, filterType, customFilter and children', () => {
    const wrapper = shallowRender({
      filter: 'foo',
      filterType: 'startsWith',
      customFilter: () => {}
    });
    expect(wrapper).toMatchSnapshot();
  });
});
