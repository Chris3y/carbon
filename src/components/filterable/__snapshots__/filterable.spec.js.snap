// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filterable provides a filterable context with filter, filterType, customFilter and children 1`] = `
<ContextProvider
  value={
    Object {
      "customFilter": [Function],
      "filter": "foo",
      "filterType": "startsWith",
    }
  }
>
  test children
</ContextProvider>
`;
